inc/audit_log_to_cloudwatch/class-cloudwatch-driver.php

```diff
@@ -142,6 +142,10 @@ public function get_records( $args ) {
 
 		$results = [ 'status' => 'Running' ];
 		while ( ! in_array( $results['status'], [ 'Failed', 'Cancelled', 'Complete' ], true ) ) {
+			// Limit how fast we poll CloudWatch via calls to getQueryResults.
+			// Queries take at a minimum 1 second, so we `sleep` before even
+			// making the first call.
+			sleep( 1 );
```

Comment:
I hate it.

Could we move the initial sleep outside the loop, and decrease the sleeping in the loop? eg 1s + 100ms in the loop

Reply:
This was intentional, as the API is never sub-secoond fast.

Reply:
OH wait, I think you re suggesting that, and it doesn't do that right now

Reply:
But, 100ms is way too little. Queries are in the seconds to 10s of second magnitude.

Reply:
Basically, I'm saying we should have a longer initial wait, and a shorter polling interval, so that we catch it as soon as it finishes without excess delay. eg if it takes 1.01s, wait 1.1s instead of 2s

Reply:
This isn't going to reduce the polling much, as the RTT is going to be 100ms+ anyway. Queries are in an order of magnitude slower, so I think 1s is an appropriate resolution.